### ISSUE TITLE
In graphql query, totalCount is incorrectly affected by limit

### ISSUE THREAD

**Issue by lonekorean** [gatsbyjs/gatsby#13342]
## Description

Per the [documentation here](https://www.gatsbyjs.org/docs/graphql-reference/#limit), `totalCount` should return the total number of existing results, even if the actual number of returned results is constrained by `limit`. I am seeing `totalCount` being constrained by `limit`.

This is an issue for pages that only want to show some of the results, but still display the total available.

### Steps to reproduce

Run this query (or something similar to it) when you have a data source with more than 3 items:

```
{
  withLimit: allMarkdownRemark(limit: 3) {
    totalCount
  }
  
  withoutLimit: allMarkdownRemark {
    totalCount
  }
}
```

You'll see 2 different `totalCount` values returned.

### Expected result

`totalCount` should not be affected by `limit`.

![expected](https://user-images.githubusercontent.com/1245573/56084673-6510f100-5e04-11e9-8b00-f81e96186379.png)

### Actual result

`totalCount` is constrained by `limit`.

![actual](https://user-images.githubusercontent.com/1245573/56084679-76f29400-5e04-11e9-9c6a-4cc4c38209ae.png)

### Environment

I tested different version of gatsby, and it seems that the bug was introduced with v2.2.0.

- v2.3.22: totalCount is **incorrect**
- v2.2.0: totalCount is **incorrect**
- v2.1.39: totalCount is **correct**

System:
    OS: Windows 10
    CPU: x64 Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz
  Binaries:
    Yarn: 1.10.1 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.1.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: 42.17134.1.0
  npmPackages:
    gatsby: ^2.1.19 => 2.3.22
    gatsby-plugin-catch-links: ^2.0.12 => 2.0.13
    gatsby-plugin-feed: ^2.0.14 => 2.1.0
    gatsby-plugin-google-analytics: ^2.0.15 => 2.0.18
    gatsby-plugin-manifest: ^2.0.20 => 2.0.29
    gatsby-plugin-offline: ^2.0.24 => 2.0.25
    gatsby-plugin-react-helmet: ^3.0.7 => 3.0.12
    gatsby-plugin-sass: ^2.0.10 => 2.0.11
    gatsby-remark-copy-linked-files: ^2.0.9 => 2.0.11
    gatsby-remark-prismjs: ^3.2.4 => 3.2.8
    gatsby-remark-smartypants: ^2.0.8 => 2.0.9
    gatsby-source-filesystem: ^2.0.23 => 2.0.29
    gatsby-transformer-json: ^2.1.8 => 2.1.11
    gatsby-transformer-remark: ^2.3.0 => 2.3.8

**Comment by polishedwp** [comment]
@lonekorean congratulations on a well put together complete Issue. It appears, based on the [Codesandbox](https://codesandbox.io/s/github/gatsbyjs/gatsby/tree/master/examples/graphql-reference), in the [documentation](https://www.gatsbyjs.org/docs/graphql-reference/#limit), that the documentation was originally put together based on Gatsby 2.0.107.

**Comment by jonniebigodes** [comment]
@lonekorean think of it this way, at the time of the writting of the documentation that was the case. As Gatsby is constantly mutating and evolving, sometimes like in this case, the documentation did not keep up. If you're willing, read the [contribution](https://www.gatsbyjs.org/contributing/how-to-contribute/) guidelines, fork the repo, make the adjustments based your reproduction and submit a pull request. Lastly great work on finding this issue and the reproduction. 👍

**Comment by KyleAMathews** [comment]
Thanks for the issue! This is actually a bug — the documentation is correct. It looks like the recent schema refactor changed the behavior incorrectly. @stefanprobst could you fix this?

**Comment by stefanprobst** [comment]
sorry for this, fixed in #13352

**Comment by lonekorean** [comment]
Thank you for the quick fix!
